### PR TITLE
libnmstate-varlink: Implementation of systemd service

### PR DESCRIPTION
**Pull request by SudhanAruna**

### CODE REVIEW

File: packaging/nmstate.spec [hunk 1]
```diff
@@ -13,6 +13,7 @@ BuildArch:      noarch
 BuildRequires:  python3-devel
 BuildRequires:  python3-setuptools
 Requires:       python3-%{libname} = %{?epoch:%{epoch}:}%{version}-%{release}
+Requires:       systemd
```
Comment (cathay4t) [inline packaging/nmstate.spec:16]: Should be `systemd-rpm-macros`

File: packaging/nmstate.spec [hunk 2]
```diff
@@ -55,13 +56,19 @@ This package contains the nmstate plugin for OVS database manipulation.
 
 %install
 %py3_install
+mkdir -p %{buildroot}%{_unitdir} && cp -a %{buildroot}%{python3_sitelib}/%{libname}/varlink/nmstate-varlink@.service %{buildroot}%{_unitdir}/
+
+%post
+%systemd_post nmstate-varlink@.service
```
Comment (cathay4t) [inline packaging/nmstate.spec:62]: Also need `%preun` and `%postun`

File: libnmstate/varlink/io.nmstate.varlink [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+method Show(include_status_data: ?bool) -> (state: ?object)
+
+method Apply(state: object, verify_change: ?bool, commit: ?bool, rollback_timeout: ?int) -> (response: ?object)
+
+method Commit(checkpoint: ?string) -> (response: ?object)
+
+method Rollback(checkpoint: ?string) -> (response: ?object)
+
+# Error definition
+error NmstateValueError (reason: object)
```
Comment (cathay4t) [inline libnmstate/varlink/io.nmstate.varlink:33]: This `object` is for `Foreign untyped/raw object`.

I think it's reasonable to assume all errors only contain string for reason.

File: libnmstate/varlink/nmstate-varlink@.service [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Requres=NetworkManager.service
+ConditionPathExists=/usr/bin/io.nmstate
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/io.nmstate %i
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
```
Comment (cathay4t) [inline libnmstate/varlink/nmstate-varlink@.service:33]: Need a new line

File: packaging/Dockerfile.centos8-nmstate-dev [hunk 5]
```diff
@@ -41,6 +41,8 @@ RUN dnf -y install dnf-plugins-core epel-release && \
 
 RUN rpm --verifydb
 
+RUN pip3 install --target=/usr/lib/python3.6/site-packages/ varlink
```
Comment (cathay4t) [inline packaging/Dockerfile.centos8-nmstate-dev:44]: Please use `python3-varlink` rpm package

File: packaging/Dockerfile.fedora-nmstate-dev [hunk 6]
```diff
@@ -46,6 +46,8 @@ RUN dnf -y install --setopt=install_weak_deps=False \
 
 RUN rpm --verifydb
 
+RUN pip3 install --target=/usr/lib/python3.6/site-packages/ varlink
```
Comment (cathay4t) [inline packaging/Dockerfile.fedora-nmstate-dev:49]: Please use `python3-varlink` rpm package

File: packaging/nmstate.spec (hunk 1 above)
Comment (SudhanAruna) [reply packaging/nmstate.spec:16]: @cathay4t In most documentations its ```systemd```. When using systemd-rpm-macros I get failed dependencies error.

File: libnmstate/varlink/nmstate-varlink@.service [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+[Unit]
+Description= Nmstate's libnmstate varlink service
+Documentation=https://www.nmstate.io/devel/libnmstate-varlink.html
+After=multi-user.target
+Requres=NetworkManager.service
+ConditionPathExists=/usr/bin/io.nmstate
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/io.nmstate %i
```
Comment (cathay4t) [inline libnmstate/varlink/nmstate-varlink@.service:29]: Only unix socket is allowed here. Due to privilege leak when using tcp socket.

You may take `/usr/lib/systemd/system/io.podman.service` as reference.

And there is no need to run nmstate as non-root user. Just make it root only.

File: packaging/nmstate.spec (hunk 1 above)
Comment (cathay4t) [reply packaging/nmstate.spec:16]: According to fedora package guide: https://docs.fedoraproject.org/en-US/packaging-guidelines/Systemd/

```
Please note that in order for the %{_unitdir} macro to exist, your package must have:

BuildRequires: systemd-rpm-macros
```

Comment (cathay4t) [reply packaging/nmstate.spec:16]: And there is no need to specify run time requirements on systemd.

This rpm spec is for Fedora and RHEL/CentOS only.

**Comment by cathay4t** [comment]
We have renamed the master branch to base. Please close this PR and create new one targeting the base branch.

Sorry for the troubles .

**Pull request closed without merging**